services/motion/builtin/move_request.go

```diff
@@ -140,6 +141,15 @@ func (mr *moveRequest) AnchorGeoPose() *spatialmath.GeoPose {
 // execute attempts to follow a given Plan starting from the index percribed by waypointIndex.
 // Note that waypointIndex is an atomic int that is incremented in this function after each waypoint has been successfully reached.
 func (mr *moveRequest) execute(ctx context.Context, plan motionplan.Plan) (state.ExecuteResponse, error) {
+	// Determine if we already are at the goal
+	// If our motion profile is position_only then, we only check against our current & desired position
+	// Conversely if our motion profile is anything else, then we also need to check again our
+	// current & desired orientation
+	resp := mr.alreadyAtGoalCheck(mr.planRequest.StartPose)
+	if resp != nil {
```

Comment:
I think the general idea behind this logic should be handled by the motion planner, not by the motion service.

That is, if the user requests a goal which is equal to (or very close to) the base/arm/whatever's current position, that should still be passed to the planner. The planner should determine that the starting state is sufficiently close to the ending state, and emit a no-op plan.

Reply:
> I think the general idea behind this logic should be handled by the motion planner, not by the motion service.

This works that way we can keep
```
spatialmath.OrientationAlmostEqual(goal.Pose().Orientation(), basePose.Orientation())
```
 instead of doing
```
spatialmath.OrientationAlmostEqualEps(goal.Pose().Orientation(), basePose.Orientation(), 5)
```